src/Login.tsx

```diff
@@ -1,16 +1,72 @@
 import './Login.css';
+import { useState } from 'react';
+import React from 'react';
 
 function Login() {
+  const [mostraError, setMostraError] = useState(false);
+  const [erros, setErros] = useState<string[]>([]);
+  const [email, setEmail] = useState('');
+  const [senha, setSenha] = useState('');
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    function validacao(email: string, senha: string): string[] {
+      const erros: string[] = [];
+
+      if (!email) {
+        erros.push('O e-mail é obrigatório.');
+      } else if (!/\S+@\S+\.\S+/.test(email)) {
+        erros.push('E-mail inválido.');
+      }
+
+      if (!senha) {
+        erros.push('A senha é obrigatória.');
+      } else if (senha.length < 7) {
+        erros.push('A senha deve ter pelo menos 7 caracteres.');
+      } else if (!/[a-zA-Z]/.test(senha) || !/\d/.test(senha)) {
+        erros.push('A senha deve ter pelo menos uma letra e um dígito.');
+      }
+
+      return erros;
+    }
+    const erros = validacao(email, senha);
+    if (erros.length > 0) {
+      setErros(erros);
+      setMostraError(true);
+    } else {
+      setMostraError(false);
+    }
```

Comment:
aqui não é necessário o `mostraErro`, apenas o estado de erro já seria suficiente e você consegue verifica no JSX se deve mostrar o erro pela quantidade de erros no estado
```suggestion
      setErros(erros);
```